### PR TITLE
Extract WinForms and WPF code to their own projects

### PR DESCRIPTION
**Pull request by Sharparam**
This will make it possible to generate a Unity build much more easily.

WPF- and WinForms-specific code is extracted into two separate projects which will have their own NuGet packages as well. The CI will include `Corale.Colore.Wpf.dll` and `Corale.Colore.WinForms.dll` in all zip distribution packages too.

Updates #157.

### CODE REVIEW

**Comment by njbmartin** [comment]
We now officially support Unity again! Tried and tested. Good work @Sharparam!

**Comment by njbmartin** [comment]
Correction, log4net does indeed cause issues:

> The class System.Diagnostics.EventLogEntryType could not be loaded, used in System, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
The class System.Configuration.IConfigurationSectionHandler could not be loaded, used in System, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089

**Comment by njbmartin** [comment]
My sincerest apologies there. After all that, I forgot to follow my own instructions.

As per http://developer.razerzone.com/chroma/download/

## Instructions for Unity

- Download the AnyCPU version to allow users on a 64-bit machine to use it, even if your game was built in 32-bit.
- Extract the files into the Assets/plugins folder within your Unity project.
- Set the API compatibility level of the player to .NET 2.0.

> By default, it is set to a subset that can’t load in the Colore Library. Head over to Edit >> Project Settings >> Player and under “Other Settings” there is a section called Optimization. You will be able to adjust the API Level below that.

File: Corale.Colore.Tester/Corale.Colore.Tester.csproj [hunk 1]
```diff
@@ -89,8 +89,28 @@
...
+    <Reference Include="Xceed.Wpf.AvalonDock.Themes.VS2010, Version=2.8.0.0, Culture=neutral, PublicKeyToken=3e4669d2f30244f4, processorArchitecture=MSIL">
+      <HintPath>..\packages\Extended.Wpf.Toolkit.2.8\lib\net40\Xceed.Wpf.AvalonDock.Themes.VS2010.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Xceed.Wpf.DataGrid, Version=2.8.0.0, Culture=neutral, PublicKeyToken=3e4669d2f30244f4, processorArchitecture=MSIL">
+      <HintPath>..\packages\Extended.Wpf.Toolkit.2.8\lib\net40\Xceed.Wpf.DataGrid.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Xceed.Wpf.Toolkit, Version=2.8.0.0, Culture=neutral, PublicKeyToken=3e4669d2f30244f4, processorArchitecture=MSIL">
+      <HintPath>..\packages\Extended.Wpf.Toolkit.2.8\lib\net40\Xceed.Wpf.Toolkit.dll</HintPath>
```
Comment (brandonscott) [inline Corale.Colore.Tester/Corale.Colore.Tester.csproj]: Are all of these needed, I cannot remember.

**Comment by brandonscott** [comment]
I've reviewed this and am happy with it.  @Sharparam I know you were looking at deployment methods to the Unity store, any progress?

@njbmartin Are you happy with this implementation?  Managed to test?

**Comment by Sharparam** [comment]
@brandonscott I haven't been able to find any information about automatic asset deployment. The easiest way for us is probably to just manually submit updates to the asset store when we build releases.

We can make the CI generate a zip file or whatever the Unity Asset Store requires and we can then just download from the CI and upload to the asset store.

**Comment by brandonscott** [comment]
@Sharparam Cool, guess we just wait for sign off from @njbmartin and then we can merge.

**Comment by njbmartin** [comment]
Yep, looks great and works well with Unity as long as you are following the existing instructions regarding the API level in Unity. Granted it doesn't officially support the subset, but as mentioned you can rectify this manually by replacing the `log4net.dll` with a Mono compatible version.

I'm just going to investigate something.

File: Corale.Colore.Tester/Corale.Colore.Tester.csproj (hunk 1 above)
Comment (Sharparam) [inline Corale.Colore.Tester/Corale.Colore.Tester.csproj]: Not sure, updating NuGet packages must've brought them in.

**Comment by njbmartin** [comment]
Let's merge this as is. It works fine when following the pre-existing instructions. We can then look into further Unity support later as per @Xinhuan's request.

I'll put a Unity guide up on the wiki.

**Pull request merged**